--- highcharts.src.js.org	2011-05-19 10:53:49.000000000 +0100
+++ highcharts.src.js	2011-05-19 10:53:04.000000000 +0100
@@ -3844,6 +3844,7 @@
 		isInsidePlot, // function
 		tooltip,
 		mouseIsDown,
+		rightClick,  // for contextmenu
 		loadingDiv,
 		loadingSpan,
 		loadingShown,
@@ -5996,17 +5997,23 @@
 		/**
 		 * Mouse up or outside the plot area
 		 */
-		function drop() {
+		function drop(e) {
+
+			e = normalizeMouseEvent(e);
+
 			if (selectionMarker) {
 				var selectionData = {
 						xAxis: [],
-						yAxis: []
+						yAxis: [],
+						// Provide the x, y location in case for context menu to display
+						// next to the pointer
+						x: e.originalEvent.x,
+						y: e.originalEvent.y
 					},
 					selectionBox = selectionMarker.getBBox(),
 					selectionLeft = selectionBox.x - plotLeft,
 					selectionTop = selectionBox.y - plotTop;
-				
-					
+
 				// a selection has been made
 				if (hasDragged) {
 					
@@ -6052,6 +6059,46 @@
 		function setDOMEvents () {
 			var lastWasOutsidePlot = true;
 			
+			// Note: Safari 5: mousedown, contextmenu, click
+			//       Firefox: mousedown contextmenu
+		        container.oncontextmenu = function(e) { 
+
+				chart.rightClick = rightClick = true;
+				//console.log('oncontextmenu');
+				
+				var hoverPoint = chart.hoverPoint;
+				e = normalizeMouseEvent(e);
+				 
+				e.cancelBubble = true; // IE specific
+				if (e.stopPropagation)
+				  e.stopPropagation();
+				if (e.preventDefault)
+				  e.preventDefault();
+
+				if (!hasDragged) {
+					if (hoverPoint && attr(e.target, 'isTracker')) {
+						var plotX = hoverPoint.plotX,
+							plotY = hoverPoint.plotY;
+							
+						// add page position info
+						extend(hoverPoint, {
+							pageX: chartPosition.left + plotLeft + 
+								(inverted ? plotWidth - plotY : plotX),
+							pageY: chartPosition.top + plotTop + 
+								(inverted ? plotHeight - plotX : plotY)
+						});
+						
+						// the series click event
+						fireEvent(hoverPoint.series, 'contextmenu', extend(e, {
+							point: hoverPoint
+						}));
+						
+						// the point click event
+						hoverPoint.firePointEvent('contextmenu', e);
+					}
+				}
+			};
+
 			/*
 			 * Record the starting position of a dragoperation
 			 */
@@ -6062,6 +6109,7 @@
 				//e.preventDefault && e.preventDefault();
 				
 				chart.mouseIsDown = mouseIsDown = true;
+				chart.rightClick = rightClick = false;
 				mouseDownX = e.chartX;
 				mouseDownY = e.chartY;
 				
@@ -6217,6 +6265,13 @@
 			
 			// MooTools 1.2.3 doesn't fire this in IE when using addEvent
 			container.onclick = function(e) {
+
+				// Safari 5: call click with contextmenu
+				if (rightClick)  {
+				  rightClick = chart.rightClick = false;
+				  return;
+				}
+
 				var hoverPoint = chart.hoverPoint;
 				e = normalizeMouseEvent(e);
 				 
